rama-dns/src/in_memory.rs

```diff
@@ -59,6 +64,16 @@ impl InMemoryDns {
         self
     }
 
+    /// Convenience method to insert an IPv4 address for a domain.
+    pub fn insert_ipv4(&mut self, name: impl Into<Domain>, addr: Ipv4Addr) -> &mut Self {
+        self.insert(name.into(), vec![IpAddr::V4(addr)])
+    }
+
+    /// Convenience method to insert an IPv6 address for a domain.
+    pub fn insert_ipv6(&mut self, name: impl Into<Domain>, addr: Ipv6Addr) -> &mut Self {
```

Comment:
```suggestion
    pub fn insert_ipv6(&mut self, name: Domain, addr: Ipv6Addr) -> &mut Self {
```

There is not a whole lot that can be turned directly into a domain as an infallible process. So might as well just expect the concrete type itself.

There is however some stuff that can be turned infallible into an Ipv4/Ipv6 addr (e.g. [u8; 4)). So might be nice addition to turn the addr part into an impl parameter for both your insert methods.

Reply:
I made it a general method. let me know if this is okay